### PR TITLE
Feature/feed

### PR DESCRIPTION
**Pull request by GarageInc**
**Summary**

Add feed with timelineEvents from followed users
Merge with historical activities

**Screenshots**
image

### CODE REVIEW

File: src/components/SignalCard/card/SignalCard.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+          isUserTheAuthor={isUserTheAuthor}
+          isPublic={isPublic}
+          removeSignal={removeSignal}
+          signal={signal}
+        />
+      )}
+
+      <div className={styles.wrapper__right}>
+        <div onClick={clickable ? goToSignalSettings : null}>
+          <div className={clickable ? styles.pointer : undefined}>
```
Comment (haritonasty) [inline src/components/SignalCard/card/SignalCard.js]:
```suggestion
          <div className={clickable && styles.pointer}>
```

File: src/components/SignalCard/card/SignalCard.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      {isMobileVersion && (
+        <DesktopOnly>
+          <SignalCardHeader
+            deleteEnabled={deleteEnabled}
+            isUserTheAuthor={isUserTheAuthor}
+            isPublic={isPublic}
+            removeSignal={removeSignal}
+            signal={signal}
+          />
+        </DesktopOnly>
```
Comment (haritonasty) [inline src/components/SignalCard/card/SignalCard.js]: Where we need to use `DesktopOnly` and have `isMobileVersion` at the same time?

File: src/components/SignalCard/card/SignalCardBottom.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            />
+          </DesktopOnly>
+        )}
+
+        {isPublished ? (
+          <h4 className={styles.author}>
+            {isAwaiting && (
+              <div className={styles.awaitingBlock}>
+                <Icon type='awaiting' />
+                <span>&nbsp;&nbsp;Awaiting confirmation</span>
```
Comment (haritonasty) [inline src/components/SignalCard/card/SignalCardBottom.js]: Do we need this case? As I know - we removed this step when you publish insight

File: src/pages/SonarFeed/SonarFeedActivityPage.js [hunk 4]
```diff
@@ -79,57 +70,31 @@ const SonarFeedActivityPage = ({ activities, isLoading, classes = {} }) => {
...
-    },
-    fetchPolicy: 'cache-and-network'
-  }),
-  props: ({ data }) => {
-    return {
-      activities: (data.activities || {}).activity,
-      isLoading: data.loading,
-      isError: !!data.error
+export const getActivitiesEnhance = () => {
+  return graphql(TRIGGER_ACTIVITIES_QUERY, {
```
Comment (haritonasty) [inline src/pages/SonarFeed/SonarFeedActivityPage.js]: you can skip `return` if you use return only in body

File: src/pages/SonarFeed/SonarFeedActivityPage.js [hunk 5]
```diff
@@ -79,57 +70,31 @@ const SonarFeedActivityPage = ({ activities, isLoading, classes = {} }) => {
...
+    options: () => ({
+      variables: makeVariables(new Date().toISOString()),
+      fetchPolicy: 'cache-and-network'
+    }),
+    props: ({ data }) => {
+      return {
+        activities: (data.activities || {}).activity,
+        loading: data.loading,
+        error: data.error
+      }
```
Comment (haritonasty) [inline src/pages/SonarFeed/SonarFeedActivityPage.js]: without return, use destructuration earlier

File: src/utils/dates.js [hunk 6]
```diff
@@ -85,7 +85,7 @@ const FormatToTimestamp = {
 }
 
 /**
- * @param {number} amount - Amount of days/months to add or substract
+ * @param {number} amount - Amount of days/months to  or substract
```
[the remark this answers is not in the document]
Comment (GarageInc) [reply src/utils/dates.js]: Wtf

**Comment by GarageInc** [comment]
> You can divide your big PR into several small ones: feed-styles, feed-trending-words, feed-loading-more
> Make it in future, please
> Very hard to read such big changes(
> 
> also, sometimes when I scroll down I get error:
> image

Will check

File: src/pages/SonarFeed/SonarFeedActivityPage.js (hunk 4 above)
Comment (GarageInc) [reply src/pages/SonarFeed/SonarFeedActivityPage.js]: @haritonasty , i know. But it is very usefull for 'debugger'. Not critical

File: src/pages/SonarFeed/SonarFeedActivityPage.js (hunk 5 above)
Comment (GarageInc) [reply src/pages/SonarFeed/SonarFeedActivityPage.js]: Same ^

File: src/components/SignalCard/card/SignalCardBottom.js (hunk 3 above)
Comment (GarageInc) [reply src/components/SignalCard/card/SignalCardBottom.js]: It is a signal card. Only moved code in separate file

File: src/components/SignalCard/card/SignalCard.js (hunk 2 above)
Comment (GarageInc) [reply src/components/SignalCard/card/SignalCard.js]: Do't remember why we are using this feature

File: src/utils/dates.js [hunk 7]
```diff
@@ -310,3 +310,15 @@ export const toEndOfDay = target => {
...
+  var result = new Date(date)
+  result.setDate(result.getDate() + days)
+  return result
+}
+
+export const dateDifferenceInWordsString = dateString => {
+  return dateDifferenceInWords({
+    from: new Date(dateString)
+  })
+}
```
[the remark this answers is not in the document]
Comment (GarageInc) [reply src/utils/dates.js]: Debugger :(

**Pull request merged**